sinker.go

```diff
@@ -74,7 +75,18 @@ func (fs *FileSinker) Run(ctx context.Context) error {
 }
 
 func (fs *FileSinker) HandleBlockScopedData(ctx context.Context, data *pbsubstreamsrpc.BlockScopedData, isLive *bool, cursor *sink.Cursor) error {
-	if err := fs.bundler.Roll(ctx, cursor.Block().Num()); err != nil {
+	var blockTime time.Time
+
+	// Extract timestamp from block data and set it on timestamp-aware writers
+	if data.Clock != nil && data.Clock.Timestamp != nil {
+		blockTime = data.Clock.Timestamp.AsTime()
+		if timestampAware, ok := fs.bundler.Writer().(writer.TimestampAware); ok {
+			timestampAware.SetCurrentTimestamp(blockTime)
+		}
+	}
```

Comment:
Timestamp must be available, when not present, an error should be returned (should fix my other comment about that which will enforce timestamp to be set).